src/isomorphic/constructRoutes.js

```diff
@@ -303,6 +308,8 @@ function setProps(element, route, htmlLayoutData) {
       throw Error(
         `Prop '${propName}' was not defined in the htmlLayoutData. Either remove this attribute from the HTML element or provide the prop's value`
       );
+    } else {
+      route.props[propName] = MISSING_PROP;
```

Comment:
On the server side, the props start out missing, and then are filled in asynchronously.